core-skills/tdd/README.md

```diff
@@ -24,91 +24,104 @@
 
-- [ ] Can identify the type of test double from code they haven’t written.
+- Can identify the type of test double from code they haven’t written.
   - Required - an explanation of a single test double in the existing test suite, including what it is and the purpose it serves in the test.
 
-- [ ] Can hand-write each of the test doubles.
+- Can hand-write each of the test doubles.
   - Required - write an example of a Dummy, a Stub, a Fake, a True Mock and a Spy, describing the purpose each serves.
 
-- [ ] Can explain when you should and shouldn’t use test doubles.
+- Can explain when you should and shouldn’t use test doubles.
   - Required - one example of when a test double is useful.
   - Required - one example of when a test double is NOT useful.
 
-- [ ] Can explain why minimizing the number of test doubles a single test case uses is desirable
+- Can explain why minimizing the number of test doubles a single test case uses is desirable
   - Required - a sensible example of how having too many test doubles can cause problems/problematic tests.
 
-- [ ] Can explain benefits of the different ways of testing remote services.
+- Can explain benefits of the different ways of testing remote services.
   - Required - Can identify three methods of remote testing.
   - Required - Can explain the pros and cons of each.
 
-- [ ] Can implement an appropriate method of testing a remote service.
+- Can implement an appropriate method of testing a remote service.
   - Required - Passing test code which handles a remote service adequately.
 
-- [ ] Can describe the situations where using triangulation helps improve test quality.
+- **DEPRECATED, BUT MUST BE ACHIEVED IF NOT DONE IN GIRAFFE** - Can describe the situations where using triangulation helps improve test quality.
   - Required - Can explain what the potential flaws of not including triangulation are.
   - Required - Can explain how triangulation overcomes these flaws.
 
-- [ ] Can demonstrate triangulation when TDDing an example piece of code.
+- **DEPRECATED, BUT MUST BE ACHIEVED IF NOT DONE IN GIRAFFE** - Can demonstrate triangulation when TDDing an example piece of code.
   - Required - Production code cannot be easily slimed, due to the use of triangulation.
 
 ### Dragon
 
 ![Dragon](./dragon.jpg)
 
-Refactor dem code.
-
-- [ ] Does not TDD markup.
-  - Required - No example of this, unless explained with good reason
-
-- [ ] Can identify smells within code they have written
-
-- [ ] Can identify smells within somebody else's code
+Review and refactor a preprepared test suite to improve performance.
 
-- [ ] Can explain the balancing act of achieving conciseness of a test suite
+- Can write an acceptance test around an existing feature.
+  - Required - Write an acceptance test around a feature of an existing application which verifies the feature works as intended.
```

Comment:
How come this is for an existing application. Implies not-test first?

We haven't defined what an acceptance test is... Is it [subcutaneous](https://martinfowler.com/bliki/SubcutaneousTest.html) or something more high-level (e.g. through the UI).

I always use Acceptance Test to refer to subcutaneous tests, and Feature Test to refer to through the UI testing.

Reply:
Agreed, it's not test first in this case. We felt that being able to demonstrate writing an acceptance test (discussed as either subcutaneous or through the UI) around an existing feature would demonstrate enough of the skills, and be a good first step in TDDing legacy code. Which is basically Dragon.

Reply:
I don't like that we have a pun on what "Acceptance Test" means at Made Tech. 

Recognising that we're talking about Subcutaneous Testing vs UI testing is an _important_ distinction.